jupyterlab_caip_optimizer/src/service/optimizer.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable @typescript-eslint/camelcase */
+import { ServerConnection } from '@jupyterlab/services';
+
+import { TransportService, handleApiError } from 'gcp_jupyterlab_shared';
+import { Study, MetadataFull, MetadataRequired } from '../types';
+
+// const AI_PLATFORM = 'https://ml.googleapis.com/v1';
+
+function zoneToRegion(zone: string): string {
+  const divider = '-';
+  const region = zone
+    .split(divider)
+    .slice(0, 2)
+    .join(divider);
+  return region;
+}
+
+/**
+ * Class to interact with Optimizer
+ */
+export class OptimizerService {
+  private readonly serverSettings = ServerConnection.defaultSettings;
+
+  constructor(private _transportService: TransportService) {}
+
+  set transportService(transportService: TransportService) {
```

Comment:
Are we planning on using this at any point?

Reply:
Not really for now, but I assumed it may be useful when we want to change the transport service some time in the future (and GCP Scheduler extension seemed to have this as well). Should I remove it?

Reply:
It looks like they use it [here](https://github.com/GoogleCloudPlatform/jupyter-extensions/blob/master/jupyterlab_gcpscheduler/src/components/dialog.tsx#L274-L281) I generally prefer only adding functions when we actually use them, but if you know that we will definitely be using it in the future I'd say it's fine to keep